packages/nimble-components/src/rich-text-mention/users/tests/rich-text-mention-users.stories.ts

```diff
@@ -3,13 +3,18 @@ import type { Meta, StoryObj } from '@storybook/html';
 import { createUserSelectedThemeStory } from '../../../utilities/tests/storybook';
 import { hiddenWrapper } from '../../../utilities/tests/hidden';
 
+const patternDescription = `A regex serves purpose of validation against the input mention markdown string. It is also used to extract the user ID from user hrefs during the parsing of markdown in the rich text components. 
+
+* To extract the user ID, the pattern should be grouped to render as mention view element if the user name is not identified in the mapping elements but matches the pattern. For example, the pattern should be \`user:(.*)\` to extract the user ID which is next to \`user:\`. If the pattern does not have grouping regex (\`user:.*\`), also if the mapping element for the particular user href is not found, the given input will render as plain text in the rich text components.
```

Comment:
Thanks for the improved description. Regarding displaying the user key instead of the display name, it occurs only when neither the pattern has a group _nor_ the mapping element is missing; it must be the absence of both conditions for rendering as plain text. I have updated the same with that correction in the doc as well.

Reply:
I'm afraid I'm still confused about the behavior and I think the description is missing some cases. Could you fill in the *Renders* column with cases like "key", "display name", "plain text or URL" for the cases below (I took a first pass but you should double check my guesses)? Then we can come up with a description that covers the cases succinctly. And then we should also ensure that we have test coverage for the relevant cases.

| Regex has group | Mapping exists with key | Display name set on mapping | Renders |
|--|--|--|--|
| Y | Y | Y | display name |
| Y | Y | N |  ? |
| Y | N | Y | N/A |
| Y | N | N |  ? |
| N | Y | Y |  ? |
| N | Y | N |  plain text or URL |
| N | N | Y |  N/A |
| N | N | N |  key |

Here's the markdown table syntax for you to copy:
```
| Regex has group | Mapping exists with key | Display name set on mapping | Renders |
|--|--|--|--|
| Y | Y | Y | display name |
| Y | Y | N |  ? |
| Y | N | Y | N/A |
| Y | N | N |  ? |
| N | Y | Y |  ? |
| N | Y | N |  plain text or URL |
| N | N | Y |  N/A |
| N | N | N |  key |
```

Reply:
Thanks for sharing the markdown syntax for the table. Here is the updated table for the below cases as per the current implementation:

| Regex has group | Mapping exists with key | Display name set on mapping | Renders |
|--|--|--|--|
| Y | Y | Y | display name |
| Y | Y | N |  plain text or URL |
| Y | N | Y | N/A (key) |
| Y | N | N |  key |
| N | Y | Y |  display name |
| N | Y | N |  plain text or URL |
| N | N | Y | N/A (plain text or URL)  |
| N | N | N |  ~key~ plain text or URL  |

In the above table, I am assuming the "key" is the user ID in `@mention` and not the entire mention-href.

The plain text or URL depends on the scheme of the autolink in markdown i.e. HTTPS/HTTP links. Also, in the second case, you might be wondering why it is "plain text or URL" instead of "key", it is because if the `display-name` is missing from the mapping element then it is a failed validation captured in mention-validator of user mentions so displaying as a link and not mention.

Reply:
Very helpful, thanks! To summarize it by grouping some of the cases, mention links will display:
1. the display name if the regex pattern matches, a mapping is found for that key, and that mapping has a display name
2. the key if the regex pattern matches and the regex has a group but no mapping is found
3. a URL or plain text otherwise, depending on whether the mention is HTTP/HTTPS

Could you update the storybook docs with a summary like that?

Reply:
I have updated this summary in the pattern description.